plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1516,10 +1542,7 @@ producer_plugin_impl::start_block_result producer_plugin_impl::start_block() {
                      itr = _unapplied_transactions.erase( itr );
                      continue;
                   }
-               } catch( const guard_exception& e ) {
-                  chain_plug->handle_guard_exception( e );
-                  return start_block_result::failed;
-               } FC_LOG_AND_DROP();
+               } LOG_AND_DROP();
```

Comment:
This change ultimately leads to the final result of `producer_plugin_impl::start_block()` being `start_block_result::succeeded` as it removes the early return.  It's possible that other calls into the controller will probably also throw and be `LOG_AND_DROP` 'ed.  That may also lead to `chain_plugin::handle_guard_exception(e);` being called multiple times before the io_service can address the graceful shutdown.  

While that is messy, I'm pretty sure we were already vulnerable to this if we threw guards while processing multiple `_pending_incoming_transactions` at the end of this method.  (each has its own internal handling/dropping of the guard exception).  

I think we need to make sure that once a guard exception is thrown we are on a path out of this method that is guaranteed not to throw/log another guard exception.  It may be more difficult to guarantee that system wide so at maybe protecting the call to `App().quit()` in `chain_plugin` with a check against `App().is_quiting()` can help prevent unexpected situations?

Reply:
It is safe to call `app.quit()` multiple times.

Reply:
Added a check for `app().is_quiting()` and return failed in that case.